core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -71,6 +82,159 @@ private[spark] class CoarseGrainedExecutorBackend(
+          if (taskUnits.nonEmpty && taskUnits.get.nonEmpty) {
+            if (userExecConfigUnits.isEmpty || userExecConfigUnits.get.isEmpty) {
+              throw new SparkException(s"Resource: $rName has units in task config " +
+                s"and executor startup config but the user specified executor resource " +
+                s"config is missing the units config - see ${userExecUnitsConfigName}.")
+            }
+            val execCountWithUnits =
+              tryConvertUnitsToBytes(execResourceInfo.count.toString, execResourceInfo.units)
+            val taskCountWithUnits =
+              tryConvertUnitsToBytes(taskCount, taskUnits.get)
+            ResourceRealCounts(execCountWithUnits, taskCountWithUnits)
+          } else {
+            throw new SparkException(
+              s"Resource: $rName has an executor units config: ${execResourceInfo.units}, but " +
+                s"the task units config is missing.")
+          }
+        } else {
+          if (taskUnits.nonEmpty && taskUnits.get.nonEmpty) {
+            throw new SparkException(
+              s"Resource: $rName has a task units config: ${taskUnits.get}, but the executor " +
+              s"units config is missing.")
+          }
+          ResourceRealCounts(execResourceInfo.count, taskCount.toLong)
+        }
+        if (realCounts.execCount < realCounts.taskCount) {
+          throw new SparkException(s"Executor resource: $rName, count: ${realCounts.execCount} " +
+            s"isn't large enough to meet task requirements of: ${realCounts.taskCount}")
+        }
+        // also make sure the executor resource count on start matches the
+        // spark.executor.resource configs specified by user
+        val userExecCountConfigName =
+          SPARK_EXECUTOR_RESOURCE_PREFIX + rName + SPARK_RESOURCE_COUNT_POSTFIX
+        val userExecConfigCount = env.conf.getOption(userExecCountConfigName).
+          getOrElse(throw new SparkException(s"Executor resource: $rName not specified " +
+            s"via config: $userExecCountConfigName, but required " +
+            s"by the task, please fix your configuration"))
+        val execConfigCountWithUnits = if (userExecConfigUnits.nonEmpty) {
+          tryConvertUnitsToBytes(userExecConfigCount, userExecConfigUnits.get)
+        } else {
+          userExecConfigCount.toLong
+        }
+        if (execConfigCountWithUnits != realCounts.execCount) {
+          throw new SparkException(s"Executor resource: $rName, count: ${realCounts.execCount} " +
+            s"doesn't match what user requests for executor count: $execConfigCountWithUnits, " +
+            s"via $userExecCountConfigName")
+        }
+      } else {
+        throw new SparkException(s"Executor resource config missing required task resource: $rName")
+      }
+    }
+  }
+
+  // visible for testing
+  def parseResources(resourcesFile: Option[String]): Map[String, ResourceInformation] = {
```

Comment:
`parseOrFindResources` because it will fallback to find

Reply:
ok